python/lsst/pipe/tasks/finalizeCharacterization.py

```diff
@@ -767,3 +798,297 @@ def compute_psf_and_ap_corr_map(self, visit, detector, exposure, src, isolated_s
+            band,
+            isolated_star_cat_dict,
+            isolated_star_source_dict
+        )
+
+        if isolated_source_table is None:
+            raise pipeBase.NoWorkFound(f'No good isolated sources found for any detectors in visit {visit}')
+
+        exposure_cat_schema = afwTable.ExposureTable.makeMinimalSchema()
+        exposure_cat_schema.addField('visit', type='L', doc='Visit number')
+
+        metadata = dafBase.PropertyList()
+        metadata.add("COMMENT", "Catalog id is detector id, sorted.")
+        metadata.add("COMMENT", "Only one detector with data has an entry.")
+
+        psf_ap_corr_cat = afwTable.ExposureCatalog(exposure_cat_schema)
+        psf_ap_corr_cat.setMetadata(metadata)
+
+        self.log.info("Starting finalizeCharacterization on detector ID %d.", detector)
+
+        psf, ap_corr_map, measured_src = self.compute_psf_and_ap_corr_map(
+            visit,
+            detector,
+            exposure,
+            src,
+            isolated_source_table,
+        )
+
+        # And now we package it together...
+        measured_src_table = None
+        if measured_src is not None:
+            record = psf_ap_corr_cat.addNew()
+            record['id'] = int(detector)
+            record['visit'] = visit
+            if psf is not None:
+                record.setPsf(psf)
+            if ap_corr_map is not None:
+                record.setApCorrMap(ap_corr_map)
+
+            measured_src['visit'][:] = visit
+            measured_src['detector'][:] = detector
+
+            measured_src_table = measured_src.asAstropy()
+
+        if measured_src_table is None:
+            raise pipeBase.NoWorkFound(f'No good sources found for visit {visit} / detector {detector}')
+
+        return pipeBase.Struct(
+            psf_ap_corr_cat=psf_ap_corr_cat,
+            output_table=measured_src_table,
+        )
+
+
+class ConsolidateFinalizeCharacterizationDetectorConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=('instrument', 'visit',),
+):
+    finalized_psf_ap_corr_detector_cats = pipeBase.connectionTypes.Input(
+        doc='Per-visit/per-detector finalized psf models and aperture corrections.',
+        name='finalized_psf_ap_corr_detector_catalog',
+        storageClass='ExposureCatalog',
+        dimensions=('instrument', 'visit', 'detector'),
+        multiple=True,
+        deferLoad=True,
+    )
+    finalized_src_detector_tables = pipeBase.connectionTypes.Input(
+        doc=('Per-visit/per-detector catalog of measurements for psf/flag/etc.'),
+        name='finalized_src_detector_table',
+        storageClass='ArrowAstropy',
+        dimensions=('instrument', 'visit', 'detector'),
+        multiple=True,
+        deferLoad=True,
+    )
+    finalized_psf_ap_corr_cat = pipeBase.connectionTypes.Output(
+        doc=('Per-visit finalized psf models and aperture corrections.  This '
+             'catalog uses detector id for the id and are sorted for fast '
```

Comment:
The ids are sorted/the catalog is sorted? Same difference I suppose, but isn't there still a source id as primary key even if it's not sorted by it?

Reply:
This is wording that John P. and I came up with ages ago, all the catalogs have it.  And if it isn't sorted then the lookup is *slow*.  So "sorted for fast" is the operative phrase.